shape-detection/resources/endtoend-helpers.js

```diff
@@ -0,0 +1,90 @@
+function imageLoadedPromise(image) {
+    return new Promise(function(resolve, reject) {
+        if (image.complete)
+            resolve();
+        image.addEventListener("load", function() {
+            resolve();
+        });
```

Comment:
I intentionally didn't do this, because I didn't want the promise to be resolved with whatever the first argument addEventListener passes to its callback, but I guess none of the callers actually use the object it's called with, so I guess it doesn't make any difference.